modules/core/src/v2/coins/utxo/recover.ts

```diff
@@ -0,0 +1,284 @@
+/**
+ * @prettier
+ */
+
+import * as _ from 'lodash';
+import * as bitcoin from '@bitgo/utxo-lib';
+import { Codes, VirtualSizes } from '@bitgo/unspents';
+
+import * as config from '../../../config';
+import { deriveKeyByPath } from '../../../bitcoin';
+import * as errors from '../../../errors';
+import { AbstractUtxoCoin, AddressInfo, UnspentInfo } from '../abstractUtxoCoin';
+
+export interface RecoverParams {
+  scan?: number;
+  userKey: string;
+  backupKey: string;
+  recoveryDestination: string;
+  krsProvider: string;
+  ignoreAddressTypes: string[];
+  bitgoKey: string;
+  walletPassphrase?: string;
+  apiKey?: string;
+  userKeyPath?: string;
+}
+
+async function queryBlockchainUnspentsPath(
+  coin: AbstractUtxoCoin,
+  params: RecoverParams,
+  keyArray: bitcoin.HDNode[],
+  basePath: string,
+  addressesById
+) {
+  const MAX_SEQUENTIAL_ADDRESSES_WITHOUT_TXS = params.scan || 20;
+  let numSequentialAddressesWithoutTxs = 0;
+
+  async function gatherUnspents(addrIndex: number) {
```

Comment:
I'd probably refactor this to be iterative instead of recursive but let's leave that for another PR